### PR TITLE
Added TS0601 _TZE204_fhvdgeuh definition

### PR DESCRIPTION
**Pull request by rfcdejong**
I bought a device and it was not supported

This one
https://nl.aliexpress.com/item/1005006780805850.html

Bought a Tuya gateway and figured out the datapoints, lots of noise, but most important datapoints are now being exposed. Couldn't figure out how to use modern extend though. Anyway it works

image

### CODE REVIEW

File: src/devices/tuya.ts [hunk 1]
```diff
@@ -11447,6 +11447,57 @@ const definitions: DefinitionWithExtend[] = [
             ],
         },
     },
+    {
+        fingerprint: tuya.fingerprint('TS0601', ['_TZE204_fhvdgeuh']),
+        model: 'TS0601',
+        vendor: 'Tuya',
+        description: 'Din rail switch with power monitoring and threshold settings',
+        fromZigbee: [tuya.fz.datapoints, fz.metering],
```
Comment (Koenkk) [inline src/devices/tuya.ts:11455]: Is `fz.metering` really needed?

File: src/devices/tuya.ts [hunk 2]
```diff
@@ -11447,6 +11447,57 @@ const definitions: DefinitionWithExtend[] = [
...
+                // [101, 'Closing recovery time (Deprecated)',tuya.valueConverter.raw],
+                [102, 'reclosing allowed times', tuya.valueConverter.raw],
+                [103, 'current_temperature', tuya.valueConverter.raw],
+                [104, 'reclosing enable', tuya.valueConverter.raw],
+                [105, 'Timer', tuya.valueConverter.raw],
+                [106, 'Cycle Schedule', tuya.valueConverter.raw],
+                [107, 'reclose recover seconds', tuya.valueConverter.raw],
+                [108, 'Random Timing', tuya.valueConverter.raw],
+                [109, 'Switch Inching', tuya.valueConverter.raw],
+                [119, 'Power-on delay power-on time', tuya.valueConverter.raw],
```
Comment (Koenkk) [inline src/devices/tuya.ts:11492]: Please use snake casing every where

```suggestion
                [119, 'power_on_delay_power_on_time', tuya.valueConverter.raw],
```

File: src/devices/tuya.ts [hunk 3]
```diff
@@ -11447,6 +11447,57 @@ const definitions: DefinitionWithExtend[] = [
...
+        vendor: 'Tuya',
+        description: 'Din rail switch with power monitoring and threshold settings',
+        fromZigbee: [tuya.fz.datapoints, fz.metering],
+        toZigbee: [tuya.tz.datapoints],
+        configure: async (device, coordinatorEndpoint) => {
+            await tuya.configureMagicPacket(device, coordinatorEndpoint);
+            // Required to get the device to start reporting
+            await device.getEndpoint(1).command('manuSpecificTuya', 'dataQuery', {});
+        },
+        exposes: [
```
Comment (Koenkk) [inline src/devices/tuya.ts:11462]: A lot of exposes are missing, make sure that add one for every datapoint

File: src/devices/tuya.ts [hunk 4]
```diff
@@ -11447,6 +11447,57 @@ const definitions: DefinitionWithExtend[] = [
             ],
         },
     },
+    {
+        fingerprint: tuya.fingerprint('TS0601', ['_TZE204_fhvdgeuh']),
+        model: 'TS0601',
```
Comment (Koenkk) [inline src/devices/tuya.ts:11452]: Is there any model number written on the device or product page? Use the same model number for the picture in https://github.com/Koenkk/zigbee2mqtt.io/pull/2994

Comment (rfcdejong) [reply src/devices/tuya.ts:11452]: Sorry, but there is no model number anywhere on the device and also not on the product page.

File: src/devices/tuya.ts (hunk 3 above)
Comment (rfcdejong) [reply src/devices/tuya.ts:11462]: I rather remove the useless datapoints, do you any datapoint that is usefull?

Maybe a button to clear the energy, but I don't know how to make an expose for that.
Same for Fault, don't know the values
Leakage current is always 0 cause I have no device that can make that higher than 0
The reclose datapoints seems useless to me

{
    "Clear energy": false,
    "Cycle Schedule": "EwAAAAAAAAAAAA==",
    "Fault": 0,
    "Leakage current": 0,
    "Phase A": {
        "0": 9,
        "1": 50,
        "2": 0,
        "3": 0,
        "4": 0,
        "5": 0,
        "6": 0,
        "7": 0
    },
    "Switch Inching": "AAAA",
    "Switch prepayment": false,
    "Timer": 0,
    "current": 0.19,
    "current_temperature": 22,
    "energy": 4.32,
    "linkquality": 0,
    "power": 27,
    "reclose recover seconds": 30,
    "reclosing allowed times": 5,
    "reclosing enable": true,
    "relay status for power on": 2,
    "state": "ON",
    "state_l1": "ON",
    "status": 1,
    "switch": true,
    "voltage": 232.4
}

File: src/devices/tuya.ts (hunk 1 above)
Comment (rfcdejong) [reply src/devices/tuya.ts:11455]: Isn't fz.metering needed for PhaseA to current, voltage and power? Or is that tuya.valueConverter.phaseVariant2 enough?

Comment (rfcdejong) [reply src/devices/tuya.ts:11455]: Guess you are right, looking at the code in fromZigbee.ts line 730. I'll remove it

File: src/devices/tuya.ts (hunk 2 above)
Comment (rfcdejong) [reply src/devices/tuya.ts]: Done

File: src/devices/tuya.ts (hunk 4 above)
Comment (rfcdejong) [reply src/devices/tuya.ts:11452]: Got the model number from generating external definition

image

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**